models/product/src/transformers.ts

```diff
@@ -0,0 +1,65 @@
+import { Reference, TReferenceGraphql } from '@commercetools-test-data/commons';
+import { Transformer } from '@commercetools-test-data/core';
+import type { TProduct, TProductRest, TProductGraphql } from './types';
+
+const transformers = {
+  default: Transformer<TProduct, TProduct>('default', {
+    buildFields: [
+      'productType',
+      'masterData',
+      'taxCategory',
+      'state',
+      'createdBy',
+      'lastModifiedBy',
+    ],
+  }),
+  rest: Transformer<TProduct, TProductRest>('rest', {
+    buildFields: ['masterData', 'createdBy', 'lastModifiedBy'],
+    replaceFields: ({ fields }) => {
+      return {
+        ...fields,
+        productType: Reference.random().typeId('product-type').buildRest(),
+        taxCategory: Reference.random().typeId('tax-category').buildRest(),
+        // TODO: transform from field when state model is available
```

Comment:
Do we need this comment anymore?

Reply:
We said we were going to set it to null for now and build from the representation when the `state` model was completed, and then transform to a reference for graphql. But I may have not understood that properly.

Reply:
You are doing the transform already - it's not going to change when the State model is available. The null check will remain because the field isn't required.

Reply:
Ahh - ok. ditched it.